compiler/optimizer/ValuePropagationCommon.cpp

```diff
@@ -1937,13 +1935,13 @@ void TR::ValuePropagation::transformArrayCopyCall(TR::Node *node)
 
    // We can now make assertions about the arguments to the arraycopy call
    //
-   addBlockConstraint(srcObjNode, TR::VPNonNullObject::create(this));
-   addBlockConstraint(dstObjNode, TR::VPNonNullObject::create(this));
+   addBlockConstraint(srcObjNode, TR::VPNonNullObject::create((TR::ValuePropagation *)this));
```

Comment:
I'm not a fan of this casting - isn't there something we can do to avoid this? At the very least these should be static_casts, but they look like self() calls which I am not a fan of. I would have hoped we could find a way to work around this...